### ISSUE TITLE
Undesirable result using EqualComAss

### ISSUE THREAD

**Issue by rogercolbeck** [maths/moodle-qtype_stack#583]
Using EqualComAss -1/4*pi*i is not correct when -pi*i/4 is. I attach a pdf of a dummy question to generate the issue and the usage.
![Screenshot from 2020-05-31 16-53-34](https://user-images.githubusercontent.com/32660531/83356662-7813b700-a35f-11ea-8093-833e04f2e766.png)

[EqualCommAss.pdf](https://github.com/maths/moodle-qtype_stack/files/4707898/EqualCommAss.pdf)

**Comment by knedlyk** [comment]
Can you provide screenshot of the structure of your question? It looks like
you have something wrong with your question or maxima works badly.

**Comment by rogercolbeck** [comment]
You can see the question in the pdf attached to the original post -- are more details needed?

**Comment by sangwinc** [comment]
Yes, I can see exactly why it would appear sensible to ask for "Equality up to commutativity and associativity" to consider that both of the following would be "the same".

p1:-1/4*%pi*%i;
p2:-(%i*%pi)/4;

However, when you look at the internal representation we have

p1 -> 1 * %i * %pi * (4^(−1)) * UNARY_MINUS
p2 ->     %i * %pi * (4^(−1)) * UNARY_MINUS

Notice "UNARY_MINUS" is a special tocan to deal with "-" and let it commute with multiplication.

However, the equivalence class formed by this test only establishes equivalence up to "Equality up to commutativity and associativity".

If you want the rule 1*x -> x then we would need another test.  E.g. a test which includes some of these trivial looking unit operations.  

Note under "Equality up to commutativity and associativity" (-x)*(-y) is not x*y.

A minimal example is

p1:-1/4*x;
p2:-x/4;

So in the interests of keeping this test as it is, (and so not breaking old questions), I don't consider this to be a "Undesirable result using EqualComAss", but rather you are asking for a new kind of test I've not yet implemented!

It has been on my list for a while.  The design I have in mind is allowing teachers to specify which sets of rules are includes/excluded to define the equivalence class...
https://github.com/maths/moodle-qtype_stack/blob/master/stack/maxima/elementary.mac
Basically, you need a user-configurable simplifier.

Chris
